Cython/Compiler/ExprNodes.py

```diff
@@ -9107,6 +9117,32 @@ def may_be_none(self):
 
     gil_message = "Constructing Python class"
 
+    def analyse_annotations(self, env):
+        from .AutoDocTransforms import AnnotationWriter
+        position = self.class_def_node.pos
+        pairs = []
+        for entry in (e for e in env.entries.values() if e.annotation):
+            strval = AnnotationWriter().write(entry.annotation)
+            # The writer will pre-pend a `u` for writing strings,
+            # but that results in a mangled string when annotations themselves are strings
+            # ("u'foo'" instead of "'foo'")
+            if entry.annotation.is_string_literal and strval[0:2] == "u'":
```

Comment:
Note this does not handle `u"…`. I've changed the `AnnotationWriter` in `master` to handle this internally, so that we do not need this brittle fix-up here.